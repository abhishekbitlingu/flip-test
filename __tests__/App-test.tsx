/**
 * @format
 */

import 'react-native';
import 'jest-fetch-mock';
import React from 'react';
import renderer from 'react-test-renderer';
import App from './../src/App';

it('renders correctly', () => {
  jest.useFakeTimers();
  const app = renderer.create(<App />).toJSON();
  expect(app).toMatchSnapshot();
});
