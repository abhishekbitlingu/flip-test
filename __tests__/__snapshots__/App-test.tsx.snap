// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  nativeID="animatedComponent"
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  Transaction List
                </Text>
              </View>
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "marginBottom": 36,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "white",
                              "flexDirection": "row",
                              "height": 48,
                              "margin": 5,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "flexDirection": "row",
                                "margin": 2,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "rgb(171,171,171)",
                                      "fontSize": 36,
                                    },
                                    undefined,
                                    Object {
                                      "fontFamily": "EvilIcons",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                              >
                                
                              </Text>
                            </View>
                            <TextInput
                              keyboardType="default"
                              onChangeText={[Function]}
                              onFocus={[Function]}
                              onSubmitEditing={[Function]}
                              placeholder="Search name, bank or nominal"
                              style={
                                Object {
                                  "flex": 1,
                                  "padding": 5,
                                }
                              }
                            />
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "marginHorizontal": 15,
                                "opacity": 1,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "rgb(238,110,65)",
                                  "fontWeight": "600",
                                  "paddingHorizontal": 5,
                                }
                              }
                            >
                              SORT
                            </Text>
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "rgb(238,110,65)",
                                    "fontSize": 20,
                                  },
                                  undefined,
                                  Object {
                                    "fontFamily": "Entypo",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                            >
                              
                            </Text>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "justifyContent": "center",
                                "paddingHorizontal": 20,
                              }
                            }
                          >
                            <ActivityIndicator
                              color="rgb(238,110,65)"
                              size="large"
                            />
                          </View>
                        </View>
                        <Modal
                          hardwareAccelerated={false}
                          transparent={true}
                          visible={false}
                        >
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "black",
                                "height": "100%",
                                "justifyContent": "center",
                                "left": 0,
                                "opacity": 0.4,
                                "position": "absolute",
                                "top": 0,
                                "width": "100%",
                              }
                            }
                          />
                          <View
                            pointerEvents="box-none"
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 1,
                                "justifyContent": "center",
                              }
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "white",
                                    "borderRadius": 3,
                                    "padding": 10,
                                  },
                                  Object {
                                    "shadowOffset": Object {
                                      "height": 1,
                                      "width": 0,
                                    },
                                    "shadowRadius": 4,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "alignItems": "flex-start",
                                    "alignSelf": "stretch",
                                    "justifyContent": "space-between",
                                  }
                                }
                              >
                                <View
                                  accessible={true}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "flexDirection": "row",
                                      "opacity": 1,
                                      "padding": 10,
                                      "paddingVertical": 15,
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "width": "70%",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "width": "15%",
                                        }
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "alignItems": "center",
                                            "borderColor": "rgb(238,110,65)",
                                            "borderRadius": 10,
                                            "borderWidth": 2,
                                            "height": 16,
                                            "justifyContent": "center",
                                            "width": 16,
                                          }
                                        }
                                      >
                                        <View
                                          style={
                                            Object {
                                              "borderColor": "rgb(238,110,65)",
                                              "borderRadius": 6,
                                              "borderWidth": 4,
                                              "height": 6,
                                              "width": 6,
                                            }
                                          }
                                        />
                                      </View>
                                    </View>
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "stretch",
                                          "width": "80%",
                                        }
                                      }
                                    >
                                      SORT
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "flexDirection": "row",
                                      "opacity": 1,
                                      "padding": 10,
                                      "paddingVertical": 15,
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "width": "70%",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "width": "15%",
                                        }
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "alignItems": "center",
                                            "borderColor": "rgb(238,110,65)",
                                            "borderRadius": 10,
                                            "borderWidth": 2,
                                            "height": 16,
                                            "justifyContent": "center",
                                            "width": 16,
                                          }
                                        }
                                      />
                                    </View>
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "stretch",
                                          "width": "80%",
                                        }
                                      }
                                    >
                                      Name A-Z
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "flexDirection": "row",
                                      "opacity": 1,
                                      "padding": 10,
                                      "paddingVertical": 15,
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "width": "70%",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "width": "15%",
                                        }
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "alignItems": "center",
                                            "borderColor": "rgb(238,110,65)",
                                            "borderRadius": 10,
                                            "borderWidth": 2,
                                            "height": 16,
                                            "justifyContent": "center",
                                            "width": 16,
                                          }
                                        }
                                      />
                                    </View>
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "stretch",
                                          "width": "80%",
                                        }
                                      }
                                    >
                                      Name Z-A
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "flexDirection": "row",
                                      "opacity": 1,
                                      "padding": 10,
                                      "paddingVertical": 15,
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "width": "70%",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "width": "15%",
                                        }
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "alignItems": "center",
                                            "borderColor": "rgb(238,110,65)",
                                            "borderRadius": 10,
                                            "borderWidth": 2,
                                            "height": 16,
                                            "justifyContent": "center",
                                            "width": 16,
                                          }
                                        }
                                      />
                                    </View>
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "stretch",
                                          "width": "80%",
                                        }
                                      }
                                    >
                                      Date Newest
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "flexDirection": "row",
                                      "opacity": 1,
                                      "padding": 10,
                                      "paddingVertical": 15,
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "width": "70%",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "width": "15%",
                                        }
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "alignItems": "center",
                                            "borderColor": "rgb(238,110,65)",
                                            "borderRadius": 10,
                                            "borderWidth": 2,
                                            "height": 16,
                                            "justifyContent": "center",
                                            "width": 16,
                                          }
                                        }
                                      />
                                    </View>
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "stretch",
                                          "width": "80%",
                                        }
                                      }
                                    >
                                      Date Oldest
                                    </Text>
                                  </View>
                                </View>
                                <View />
                              </View>
                            </View>
                          </View>
                        </Modal>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
